pkg/catalog/uncommitted_iterator.go

```diff
@@ -0,0 +1,133 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/treeverse/lakefs/pkg/graveler"
+)
+
+type UncommittedIterator struct {
+	store     Store
+	ctx       context.Context
+	err       error
+	branchItr graveler.BranchIterator
+	entryItr  *valueEntryIterator
+	entry     *UncommittedRecord
+}
+
+type UncommittedRecord struct {
+	branchID graveler.BranchID
+	*EntryRecord
+}
+
+func NewUncommittedIterator(ctx context.Context, store Store, repository *graveler.RepositoryRecord) (*UncommittedIterator, error) {
```

Comment:
Do we have a test for this file?

Reply:
As mentioned before - due to the size of the PR the tests for the iterator will be deferred to another task